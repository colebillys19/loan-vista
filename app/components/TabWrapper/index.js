/**
 * TabWrapper
 * @description ...
 */

import styled from 'styled-components';

export default styled.section`
  background-color: white;
  display: inline-block;
  margin-bottom: 4rem;
  min-height: 116rem;
  vertical-align: top;
  width: 83.7rem;
`;
