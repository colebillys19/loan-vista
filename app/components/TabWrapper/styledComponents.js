import styled from 'styled-components';

import { appColorA } from 'styleConstants';

export const StyledSection = styled.section`
  background-color: white;
  display: inline-block;
  margin-bottom: 4rem;
  min-height: 116rem;
  vertical-align: top;
  width: 83.7rem;
  padding-top: 1rem;
  border-top: 0.3rem solid ${appColorA};
`;
