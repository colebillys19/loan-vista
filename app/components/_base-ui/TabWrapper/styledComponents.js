import styled from 'styled-components';

import { appColorA, borderGrey, backgroundColor } from 'styleConstants';

export const BottomBorder = styled.div`
  background: linear-gradient(
    90deg,
    ${backgroundColor} 0%,
    ${borderGrey} 50%,
    ${backgroundColor} 100%
  );
  height: 0.1rem;
  left: 0;
  position: absolute;
  bottom: -0.1rem;
  width: 83.7rem;
`;

export const StyledSection = styled.section`
  background: ${backgroundColor};
  background: linear-gradient(
    90deg,
    ${backgroundColor} 0%,
    #fff 50%,
    ${backgroundColor} 100%
  );
  border-top: 0.3rem solid ${appColorA};
  display: inline-block;
  margin-bottom: 4rem;
  min-height: 112rem;
  position: relative;
  vertical-align: top;
  width: 83.7rem;
`;
