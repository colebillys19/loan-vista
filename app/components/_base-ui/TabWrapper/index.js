/**
 * TabWrapper
 * @description ...
 */

import React from 'react';
import T from 'prop-types';

import { BottomBorder, StyledSection } from './styledComponents';

const TabWrapper = ({ children }) => (
  <StyledSection>
    {children}
    <BottomBorder />
  </StyledSection>
);

TabWrapper.propTypes = { children: T.node.isRequired };

export default TabWrapper;
