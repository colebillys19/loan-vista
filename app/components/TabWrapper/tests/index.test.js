import React from 'react';
import { render } from 'react-testing-library';

import TabWrapper from '../index';

describe('<TabWrapper />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<TabWrapper />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<TabWrapper />);
    expect(firstChild).toMatchSnapshot();
  });
});
