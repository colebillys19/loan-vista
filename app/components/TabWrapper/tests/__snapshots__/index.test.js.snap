// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabWrapper /> Should render and match the snapshot 1`] = `
<section
  class="TabWrapper-j2wsfb-0 beIMbY"
/>
`;
